packages/strapi-plugin-content-manager/admin/src/components/DynamicZone/index.js

```diff
@@ -3,67 +3,40 @@ import { get } from 'lodash';
 import isEqual from 'react-fast-compare';
 import PropTypes from 'prop-types';
 import { FormattedMessage } from 'react-intl';
-import { Arrow } from '@buffetjs/icons';
-import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import pluginId from '../../pluginId';
-import useEditView from '../../hooks/useEditView';
-import DynamicComponentCard from '../DynamicComponentCard';
-import FieldComponent from '../FieldComponent';
 import NotAllowedInput from '../NotAllowedInput';
 import connect from './utils/connect';
 import select from './utils/select';
 import BaselineAlignement from './BaselineAlignement';
 import Button from './Button';
-import ComponentsPicker from './ComponentsPicker';
+import Component from './Component';
 import ComponentWrapper from './ComponentWrapper';
 import DynamicZoneWrapper from './DynamicZoneWrapper';
 import Label from './Label';
-import RoundCTA from './RoundCTA';
 import Wrapper from './Wrapper';
+import Picker from './Picker';
 
 /* eslint-disable react/no-array-index-key */
 
 const DynamicZone = ({
-  max,
-  min,
   name,
-
   // Passed with the select function
   addComponentToDynamicZone,
   formErrors,
   isCreatingEntry,
   isFieldAllowed,
   isFieldReadable,
-  layout,
   moveComponentUp,
   moveComponentDown,
   removeComponentFromDynamicZone,
   dynamicDisplayedComponents,
+  fieldSchema,
+  metadatas,
 }) => {
   const [isOpen, setIsOpen] = useState(false);
 
-  const { components } = useEditView();
-
-  const getDynamicComponentSchemaData = useCallback(
-    componentUid => {
-      const component = components.find(compo => compo.uid === componentUid);
-      const { schema } = component;
-
-      return schema;
-    },
-    [components]
-  );
-
-  const getDynamicComponentInfos = useCallback(
-    componentUid => {
-      const {
-        info: { icon, name },
-      } = getDynamicComponentSchemaData(componentUid);
-
-      return { icon, name };
-    },
-    [getDynamicComponentSchemaData]
-  );
+  // We cannot use the default props here
+  const { max = Infinity, min = -Infinity } = fieldSchema;
```

Comment:
Because we forbid users from adding more components than the max limit validations. If I don't set it to Infinity if it doesn't exist the user cannot add a single one.